modeshape-jcr/src/main/java/org/modeshape/jcr/txn/NoClientTransactions.java

```diff
@@ -50,14 +50,15 @@ public Transaction currentTransaction() {
 
     @Override
     public synchronized Transaction begin() throws NotSupportedException, SystemException {
-        if (ACTIVE_TRANSACTION.get() == null) {
+        NestableThreadLocalTransaction tx = ACTIVE_TRANSACTION.get();
+        if (tx == null) {
             // Start a transaction ...
             txnMgr.begin();
             if (logger.isTraceEnabled()) {
                 logger.trace("Begin transaction {0}", currentTransactionId());
             }
-            ACTIVE_TRANSACTION.set(new NestableThreadLocalTransaction(txnMgr, ACTIVE_TRANSACTION));
+            tx = new NestableThreadLocalTransaction(txnMgr, ACTIVE_TRANSACTION);
         }
-        return ACTIVE_TRANSACTION.get().begin();
+        return tx.begin();
```

Comment:
I'm not sure I understand these changes ... the thread-local `ACTIVE_TRANSACTION` is checked above, but it is no longer ever set within this class. (And this class is final, so there can be no subclasses that do set it.)

Either `ACTIVE_TRANSACTION` should be removed and this code cleaned up, or it should be fixed so that something sets it with a `NestableThreadLocalTransaction` reference.